fabric-model-loading-api-v1/src/client/java/net/fabricmc/fabric/api/client/model/loading/v1/ModelModifier.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.model.loading.v1;
+
+import net.minecraft.client.render.model.BakedModel;
+import net.minecraft.client.render.model.Baker;
+import net.minecraft.client.render.model.ModelBakeSettings;
+import net.minecraft.client.render.model.ModelLoader;
+import net.minecraft.client.render.model.UnbakedModel;
+import net.minecraft.client.texture.Sprite;
+import net.minecraft.client.util.SpriteIdentifier;
+import net.minecraft.util.Identifier;
+
+import java.util.function.Function;
+
+/**
+ * Contains interfaces for the events mods can use to modify models. These events have multiple phases: OVERRIDE,
```

Comment:
This is a good start, but I think it would be good to expand the javadoc a bit, and give some instructions on the expected use cases. I could imagine a few paragraphs:

- Intro line - current sentence is fine.
- Use case example 1: overriding a model for a specific block state for example (just describing it in a few words is fine: one needs to check the identifier and replace the model).
- Use case example 2: wrapping a model to perform modifications.
- Explain why phases are needed to make sure that model overrides happen early and model wrapping happens late. Could also link `Event#register(Identifier, Object)` as a reminder on how to use the phases.